src/main/webapp/blur-helpers.js

```diff
@@ -259,3 +273,122 @@ async function loadBlurredPhoto(photo) {
   };
   return getImageWithBlurredAreas(imageDetails, blurRadius);
 }
+
+/**
+ * Function that preloads some of the photos in sessionStorage.
+ * Assigns each photo a value and uses the Knapsack algorithm to decide which
+ * photos to load in maximum CACHE_SIZE space so that the sum of their value is
+ * the biggest.
+*/
+async function preloadPhotos() {
+  // Fetch photos information from the server.
+  const photosResponse = await fetch('/photos');
+  const photos = await photosResponse.json();
+
+  if (photos.length === 0) {
+    return;
+  }
+
+  // Assign each photo a value.
+  const serverTimeResponse = await fetch('server-time');
+  const currentDate = Date.parse(await serverTimeResponse.text());
+  for (const photo of photos) {
+    // We divide by 1000 to convert from miliseconds to seconds.
+    const secondsOld = (currentDate - Date.parse(photo.dateCreated)) / 1000;
+    photo.value = 1 / secondsOld;
```

Comment:
To protect yourself against possible quirks of operations with time you can check that secondsOld is a positive number. Also what's the rationale of converting milliseconds to seconds? Can you have the value = 1 / millisecondsOld?

Reply:
In my opinion checking if secondsOld is positive is not worth it because we subtract an older date from the current date. It may be not positive because of a bug somewhere, but I don't think this scenario is likely to happen, as there is only one operation with values taken directly from the server.

Converting to seconds was purely for a better debugging (it's easier to calculate manually the values). But yes, changing it to milliseconds is a good idea, thank you :)